TypeScript "trailing comma not allowed" error in type parameters

Source:

```ts
export class BaseSingleLevelProfileTargeting <T extends ValidSingleLevelProfileNode> {
```

Prettier output:

```ts
export class BaseSingleLevelProfileTargeting<
	T extends ValidSingleLevelProfileNode,
> {
```

TypeScript (2.3.3) compiler when compiling the prettier output:

```ts
T extends ValidSingleLevelProfileNode,
                                     ~

(59,39): error TS1009: Trailing comma not allowed.
```

Strangely typescript playground does not seem to have a problem with it:
http://www.typescriptlang.org/play/#src=type%20ValidSingleLevelProfileNode%20%3D%20any%0D%0A%0D%0Aexport%20class%20BaseSingleLevelProfileTargeting%3C%0D%0A%20%20%20%20T%20extends%20ValidSingleLevelProfileNode%2C%0D%0A%3E%20%7B%20%7D

Although that is logging `ts.version: 2.1.4-insiders.20161201` to the console, so perhaps the behaviour changed at some point between those two TypeScript versions?

@danielrosenwasser do you know if that is the case?

--- a/src/printer.js
+++ b/src/printer.js
@@ -3523,7 +3523,11 @@ function printTypeParameters(path, options, print, paramsKey) {
           join(concat([",", line]), path.map(print, paramsKey))
         ])
       ),
-      ifBreak(shouldPrintComma(options, "all") ? "," : ""),
+      ifBreak(
+        options.parser !== "typescript" && shouldPrintComma(options, "all")
+          ? ","
+          : ""
+      ),
       softline,
       ">"
     ])

--- a/tests/typescript_trailing_comma/jsfmt.spec.js
+++ b/tests/typescript_trailing_comma/jsfmt.spec.js
@@ -1,0 +1,1 @@
+run_spec(__dirname, { parser: "typescript", trailingComma: "all" });

--- a/tests/typescript_trailing_comma/trailing.ts
+++ b/tests/typescript_trailing_comma/trailing.ts
@@ -1,0 +1,4 @@
+export class BaseSingleLevelProfileTargeting<
+	T extends ValidSingleLevelProfileNode,
+> {
+}

